### PR TITLE
Integration tests for stake API

### PR DESCRIPTION
**Pull request by pgarg66**
#### Problem
Missing integration tests for stake API

#### Summary of Changes
Added some tests

### CODE REVIEW

**Comment by pgarg66** [comment]
@rob-solana seems the circular dependency is mainly because of genesis_utils. Can it be moved out of runtime?

File: programs/stake_tests/tests/stake_instruction.rs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            100,
+        )],
+        Some(&mint_pubkey),
+    );
+    assert!(bank_client
+        .send_message(&[&mint_keypair, &staker_keypair], message)
+        .is_err());
+
+    // Create a new bank at later epoch (within cooldown period)
+    let mut bank = Bank::new_from_parent(&bank, &Pubkey::default(), 2 + bank.slot());
```
Comment (rob-solana) [inline programs/stake_tests/tests/stake_instruction.rs]: assert something about what you think you know about this bank's epoch

File: programs/stake_tests/tests/stake_instruction.rs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+use solana_sdk::signature::{Keypair, KeypairUtil};
+use solana_stake_api::id;
+use solana_stake_api::stake_instruction;
+use solana_stake_api::stake_instruction::process_instruction;
+use solana_stake_api::stake_state::StakeState;
+use solana_vote_api::vote_instruction;
+use std::sync::Arc;
+
+#[test]
+fn test_stake_account_delegate() {
```
Comment (rob-solana) [inline programs/stake_tests/tests/stake_instruction.rs]: missing is redemption...

Comment (pgarg66) [reply programs/stake_tests/tests/stake_instruction.rs]: Done

File: programs/stake_tests/tests/stake_instruction.rs (hunk 1 above)
Comment (pgarg66) [reply programs/stake_tests/tests/stake_instruction.rs]: Done

**Comment by pgarg66** [comment]
@rob-solana is it good to go?

File: programs/stake_tests/tests/stake_instruction.rs [hunk 3]
```diff
@@ -80,11 +87,105 @@ fn test_stake_account_delegate() {
...
+        ));
+
+        let bank_client = BankClient::new_shared(&bank);
+
+        let message = Message::new_with_payer(
+            vec![vote_instruction::vote(
+                &vote_pubkey,
+                &vote_pubkey,
+                vec![Vote::new(
+                    (bank.slot() - 1) as u64,
```
Comment (rob-solana) [inline programs/stake_tests/tests/stake_instruction.rs]: could use bank.parent().slot(), or take slot and hash at the top (after calling bank.freeze() on the parent)

File: programs/stake_tests/tests/stake_instruction.rs [hunk 4]
```diff
@@ -80,11 +87,105 @@ fn test_stake_account_delegate() {
...
+    let vote_state: VoteState = account.state().expect("couldn't unpack account data");
+
+    // 1 less vote, as the first vote should have cleared the lockout
+    assert_eq!(vote_state.votes.len(), 31);
+    assert_eq!(vote_state.credits(), 1);
+    assert_ne!(old_epoch, bank.epoch());
+    old_epoch = bank.epoch();
+
+    // Cycle thru banks until we reach next epoch
+    while bank.epoch() != old_epoch + 1 {
```
Comment (rob-solana) [inline programs/stake_tests/tests/stake_instruction.rs]: can you de-copy-pasta these 2 identical blocks?

**Comment by pgarg66** [comment]
> looks good, except for some nits... so: I'm dying to know: how much did the stake earn those 2 epochs?

The reward was about 19400. It got split evenly into vote and stake account..

**Pull request merged**